src/charm.py

```diff
@@ -189,11 +202,16 @@ def _restart(self, event: EventBase) -> None:
 
         self.snap.restart_snap_service("kafka")
 
-    def _set_password_action(self, event: ActionEvent):
+    def _set_password_action(self, event: ActionEvent) -> None:
         """Handler for set-password action.
 
         Set the password for a specific user, if no passwords are passed, generate them.
         """
+        if not self.peer_relation:
```

Comment:
what if someone wants a single unit of kafka? shouldn't they be able to reset password